services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -229,9 +254,24 @@ public TopicManagementService(Map<String, Object> props, String serviceName) {
     _topic = config.getString(CommonServiceConfig.TOPIC_CONFIG);
     _partitionsToBrokerRatio = config.getDouble(CommonServiceConfig.PARTITIONS_TO_BROKER_RATO_CONFIG);
     _scheduleIntervalMs = config.getInt(TopicManagementServiceConfig.REBALANCE_INTERVAL_MS_CONFIG);
-    String zkUrl = config.getString(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
-    _zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+    _zkConnect = config.getString(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    _zkUtils = ZkUtils.apply(_zkConnect, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
     _executor = Executors.newSingleThreadScheduledExecutor(new TopicManagementServiceThreadFactory());
+    _configuredReplicationFactor = config.getInt(CommonServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
+    _topicCreationEnabled = config.getBoolean(CommonServiceConfig.TOPIC_CREATION_ENABLED_CONFIG);
+
+    int existingPartitionCount = Utils.getPartitionNumForTopic(_zkConnect, _topic);
```

Comment:
It is not good to duplicate code in the project. Maybe it can be moved to Utils.java.